tests/test_plane.h

```diff
@@ -0,0 +1,183 @@
+
+	CHECK_MESSAGE(
+			plane.normalized().is_equal_approx(plane_normalized),
+			"normalized() should return a copy of the normalized value.");
+}
+
+TEST_CASE("[Plane] Basic setters") {
+	Plane plane = Plane(32, 22, 16, 3);
+	plane.set_normal(Vector3(4, 2, 3));
+
+	CHECK_MESSAGE(
+			plane.is_equal_approx(Plane(4, 2, 3, 3)),
+			"set_normal() should result in the expected Plane.");
+
+	plane = Plane(32, 22, 16, 3);
+	plane.normalize();
+
+	CHECK_MESSAGE(
+			plane.is_equal_approx(Plane(32.0 / 42, 22.0 / 42, 16.0 / 42, 3.0 / 42)),
+			"normalize() should result in the expected plane.");
+}
+
+TEST_CASE("[Plane] Plane-Point operations") {
+	const Plane plane = Plane(32, 22, 16, 3);
+	const Plane y_facing_plane = Plane(0, 1, 0, 4);
+
+	CHECK_MESSAGE(
+			plane.center().is_equal_approx(Vector3(32 * 3, 22 * 3, 16 * 3)),
+			"center() should return a vector pointing to the center of the plane.");
+
+	CHECK_MESSAGE(
+			y_facing_plane.is_point_over(Vector3(0, 5, 0)),
+			"is_point_over() should return the expected result");
+
+	CHECK_MESSAGE(
+			y_facing_plane.get_any_perpendicular_normal().is_equal_approx(Vector3(1, 0, 0)),
+			"get_any_perpindicular_normal() should return the expected result.");
+
+	// TODO distance_to()
+}
+
+TEST_CASE("[Plane] Has Point") {
+	const Plane x_facing_plane = Plane(1, 0, 0, 0);
+	const Plane y_facing_plane = Plane(0, 1, 0, 0);
+	const Plane z_facing_plane = Plane(0, 0, 1, 0);
+
+	const Vector3 x_axis_point = Vector3(10, 0, 0);
+	const Vector3 y_axis_point = Vector3(0, 10, 0);
+	const Vector3 z_axis_point = Vector3(0, 0, 10);
+
+	const Plane x_facing_plane_with_d_offset = Plane(1, 0, 0, 1);
+	const Vector3 y_axis_point_with_d_offset = Vector3(1, 10, 0);
+
+	CHECK_MESSAGE(
+			x_facing_plane.has_point(y_axis_point),
+			"has_point with contained Vector3 should return the expected result.");
+	CHECK_MESSAGE(
+			x_facing_plane.has_point(z_axis_point),
+			"has_point with contained Vector3 should return the expected result.");
+
+	CHECK_MESSAGE(
+			y_facing_plane.has_point(x_axis_point),
+			"has_point with contained Vector3 should return the expected result.");
+	CHECK_MESSAGE(
+			y_facing_plane.has_point(z_axis_point),
+			"has_point with contained Vector3 should return the expected result.");
+
+	CHECK_MESSAGE(
+			z_facing_plane.has_point(y_axis_point),
+			"has_point with contained Vector3 should return the expected result.");
+
+	CHECK_MESSAGE(
+			z_facing_plane.has_point(x_axis_point),
+			"has_point with passed Vector3 should return the expected result.");
+
+	CHECK_MESSAGE(
+			x_facing_plane_with_d_offset.has_point(y_axis_point_with_d_offset),
+			"has_point with passed Vector3 should return the expected result.");
+}
+
+TEST_CASE("[Plane] Intersection") {
+	const Plane x_facing_plane = Plane(1, 0, 0, 1);
+	const Plane y_facing_plane = Plane(0, 1, 0, 2);
+	const Plane z_facing_plane = Plane(0, 0, 1, 3);
+
+	// Intersection functions need a pointer of a vector to modify
+	Vector3 *vec_out = new Vector3();
```

Comment:
Can be rewritten to:
```suggestion
	Vector3 vec_out;
```
And then use the variable by passing an address (`&vec_out`):
```
plane.intersect_3(y_facing_plane, z_facing_plane, &vec_out),
```
This way, there's no need to worry about `delete` afterward.

Also, if you do need to use `new`, Godot uses `memnew` macro instead (along with `memdelete`), used to better track memory.